java.lang.IllegalArgumentException: Error building function signature with range 586 - 118 for element

<!-- The bug you're experiencing might have already be reported! -->
<!-- Please search in the [issues](https://github.com/detekt/detekt/issues) before creating one. -->

## Expected Behavior
Detekt runs on all expression body methods

## Observed Behavior
Detekt is crashing on certain expression body methods

## Steps to Reproduce
The exception occurred when the method was written as an expression body and when rewriting it as a block body, it no longer occurred. 

## Context
A developer was editing a file, and another file began causing detekt to throw an exception with the given stack trace below.  The file throwing the exception which did not change was passing detekt previously as we run detekt and ensure everything is baselined or suppressed before allowing a commit. 

I've edited the original error to remove the source code, but the method in question takes three parameters and returns a boolean.  The contents of the method are an `if / else` block. 

Please let me know if there is anything else I can provide. 

## Your Environment
<!-- Include as many relevant details about the environment you experienced the bug in -->
* Version of detekt used: 1.17.1
* Version of Gradle used (if applicable): N/A
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: We see this issue on both Mac and Ubuntu. 
* Link to your project (if it's a public repository): N/A

## Stack Trace

> java.util.concurrent.CompletionException: java.lang.IllegalStateException: Analyzing [FilePath.kt] led to an exception. 
>  The original exception message was: java.lang.IllegalArgumentException: Error building function signature with range 586 - 118 for element: [Method and source code]
>  Running detekt '1.17.1' on Java '1.8.0_172-b11' on OS 'Linux'
>  If the exception message does not help, please feel free to create an issue on our GitHub page.
>  	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
>  	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
>  	at java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:874)
>  	at java.util.concurrent.CompletableFuture$UniExceptionally.tryFire(CompletableFuture.java:852)
>  	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
>  	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1595)
>  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
>  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
>  	at java.lang.Thread.run(Thread.java:748)
>  Caused by: java.lang.IllegalStateException: Analyzing [FilePath.kt] led to an exception. 
>  The original exception message was: java.lang.IllegalArgumentException: Error building function signature with range 586 - 118 for element: [Method and source code]
>  Running detekt '1.17.1' on Java '1.8.0_172-b11' on OS 'Linux'
>  If the exception message does not help, please feel free to create an issue on our GitHub page.
>  	at io.gitlab.arturbosch.detekt.core.AnalyzerKt.throwIllegalStateException(Analyzer.kt:152)
>  	at io.gitlab.arturbosch.detekt.core.AnalyzerKt.access$throwIllegalStateException(Analyzer.kt:1)
>  	at io.gitlab.arturbosch.detekt.core.Analyzer$runAsync$tasks$1$2.invoke(Analyzer.kt:87)
>  	at io.gitlab.arturbosch.detekt.core.Analyzer$runAsync$tasks$1$2.invoke(Analyzer.kt:30)
>  	at io.gitlab.arturbosch.detekt.core.TaskPoolKt$sam$java_util_function_Function$0.apply(TaskPool.kt)
>  	at java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:870)
>  	... 6 more
>  Caused by: java.util.concurrent.CompletionException: java.lang.IllegalArgumentException: Error building function signature with range 586 - 118 for element: [Method and source code]
>  	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
>  	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
>  	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1592)
>  	... 3 more
>  Caused by: java.lang.IllegalArgumentException: Error building function signature with range 586 - 118 for element: [Method and source code]
>  	at io.gitlab.arturbosch.detekt.api.internal.SignaturesKt.buildFunctionSignature(Signatures.kt:108)
>  	at io.gitlab.arturbosch.detekt.api.internal.SignaturesKt.searchSignature(Signatures.kt:71)
>  	at io.gitlab.arturbosch.detekt.api.internal.SignaturesKt.buildFullSignature(Signatures.kt:46)
>  	at io.gitlab.arturbosch.detekt.api.Entity$Companion.from(Entity.kt:74)
>  	at io.gitlab.arturbosch.detekt.api.Entity$Companion.from(Entity.kt:63)
>  	at io.gitlab.arturbosch.detekt.api.Entity$Companion.from(Entity.kt:44)
>  	at io.gitlab.arturbosch.detekt.api.Entity$Companion.from$default(Entity.kt:42)
>  	at io.gitlab.arturbosch.detekt.rules.documentation.EndOfSentenceFormat.verify(EndOfSentenceFormat.kt:49)
>  	at io.gitlab.arturbosch.detekt.rules.documentation.KDocStyle.visitDeclaration(KDocStyle.kt:20)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitDeclaration(KtVisitorVoid.java:453)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitDeclaration(KtVisitorVoid.java:21)
>  	at org.jetbrains.kotlin.psi.KtVisitor.visitNamedDeclaration(KtVisitor.java:398)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitNamedDeclaration(KtVisitorVoid.java:381)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitNamedDeclaration(KtVisitorVoid.java:965)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitNamedDeclaration(KtVisitorVoid.java:21)
>  	at org.jetbrains.kotlin.psi.KtVisitor.visitNamedFunction(KtVisitor.java:53)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitNamedFunction(KtVisitorVoid.java:49)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitNamedFunction(KtVisitorVoid.java:483)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitNamedFunction(KtVisitorVoid.java:21)
>  	at org.jetbrains.kotlin.psi.KtNamedFunction.accept(KtNamedFunction.java:51)
>  	at org.jetbrains.kotlin.psi.KtElementImplStub.accept(KtElementImplStub.java:60)
>  	at org.jetbrains.kotlin.com.intellij.psi.impl.PsiElementBase.acceptChildren(PsiElementBase.java:69)
>  	at org.jetbrains.kotlin.psi.KtTreeVisitorVoid.visitElement(KtTreeVisitorVoid.java:25)
>  	at org.jetbrains.kotlin.psi.KtVisitor.visitKtElement(KtVisitor.java:24)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtElement(KtVisitorVoid.java:25)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtElement(KtVisitorVoid.java:447)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtElement(KtVisitorVoid.java:21)
>  	at org.jetbrains.kotlin.psi.KtVisitor.visitClassBody(KtVisitor.java:98)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitClassBody(KtVisitorVoid.java:89)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitClassBody(KtVisitorVoid.java:537)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitClassBody(KtVisitorVoid.java:21)
>  	at org.jetbrains.kotlin.psi.KtClassBody.accept(KtClassBody.kt:38)
>  	at org.jetbrains.kotlin.psi.KtElementImplStub.accept(KtElementImplStub.java:60)
>  	at org.jetbrains.kotlin.com.intellij.psi.impl.PsiElementBase.acceptChildren(PsiElementBase.java:69)
>  	at org.jetbrains.kotlin.psi.KtTreeVisitorVoid.visitElement(KtTreeVisitorVoid.java:25)
>  	at org.jetbrains.kotlin.psi.KtVisitor.visitKtElement(KtVisitor.java:24)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtElement(KtVisitorVoid.java:25)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtElement(KtVisitorVoid.java:447)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtElement(KtVisitorVoid.java:21)
>  	at org.jetbrains.kotlin.psi.KtVisitor.visitExpression(KtVisitor.java:182)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitExpression(KtVisitorVoid.java:169)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitExpression(KtVisitorVoid.java:659)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitExpression(KtVisitorVoid.java:21)
>  	at org.jetbrains.kotlin.psi.KtVisitor.visitDeclaration(KtVisitor.java:29)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitDeclaration(KtVisitorVoid.java:29)
>  	at io.gitlab.arturbosch.detekt.rules.documentation.KDocStyle.visitDeclaration(KDocStyle.kt:18)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitDeclaration(KtVisitorVoid.java:453)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitDeclaration(KtVisitorVoid.java:21)
>  	at org.jetbrains.kotlin.psi.KtVisitor.visitNamedDeclaration(KtVisitor.java:398)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitNamedDeclaration(KtVisitorVoid.java:381)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitNamedDeclaration(KtVisitorVoid.java:965)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitNamedDeclaration(KtVisitorVoid.java:21)
>  	at org.jetbrains.kotlin.psi.KtVisitor.visitClassOrObject(KtVisitor.java:41)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitClassOrObject(KtVisitorVoid.java:37)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitClassOrObject(KtVisitorVoid.java:465)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitClassOrObject(KtVisitorVoid.java:21)
>  	at org.jetbrains.kotlin.psi.KtVisitor.visitClass(KtVisitor.java:33)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitClass(KtVisitorVoid.java:33)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitClass(KtVisitorVoid.java:459)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitClass(KtVisitorVoid.java:21)
>  	at org.jetbrains.kotlin.psi.KtClass.accept(KtClass.kt:20)
>  	at org.jetbrains.kotlin.psi.KtElementImplStub.accept(KtElementImplStub.java:60)
>  	at org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.SharedImplUtil.acceptChildren(SharedImplUtil.java:185)
>  	at org.jetbrains.kotlin.com.intellij.psi.impl.source.PsiFileImpl.acceptChildren(PsiFileImpl.java:753)
>  	at org.jetbrains.kotlin.psi.KtTreeVisitorVoid.visitElement(KtTreeVisitorVoid.java:25)
>  	at org.jetbrains.kotlin.com.intellij.psi.PsiElementVisitor.visitFile(PsiElementVisitor.java:35)
>  	at org.jetbrains.kotlin.psi.KtVisitor.visitKtFile(KtVisitor.java:73)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtFile(KtVisitorVoid.java:69)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtFile(KtVisitorVoid.java:513)
>  	at org.jetbrains.kotlin.psi.KtVisitorVoid.visitKtFile(KtVisitorVoid.java:21)
>  	at org.jetbrains.kotlin.psi.KtFile.accept(KtFile.kt:243)
>  	at org.jetbrains.kotlin.psi.KtFile.accept(KtFile.kt:230)
>  	at io.gitlab.arturbosch.detekt.api.internal.BaseRule.visit(BaseRule.kt:61)
>  	at io.gitlab.arturbosch.detekt.api.internal.BaseRule.visitFile(BaseRule.kt:51)
>  	at io.gitlab.arturbosch.detekt.core.Analyzer$analyze$4.invoke(Analyzer.kt:121)
>  	at io.gitlab.arturbosch.detekt.core.Analyzer.analyze(Analyzer.kt:133)
>  	at io.gitlab.arturbosch.detekt.core.Analyzer.access$analyze(Analyzer.kt:30)
>  	at io.gitlab.arturbosch.detekt.core.Analyzer$runAsync$$inlined$map$lambda$1.invoke(Analyzer.kt:84)
>  	at io.gitlab.arturbosch.detekt.core.Analyzer$runAsync$$inlined$map$lambda$1.invoke(Analyzer.kt:30)
>  	at io.gitlab.arturbosch.detekt.core.TaskPoolKt$task$1.get(TaskPool.kt:13)
>  	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
>  	... 3 more
>

--- a/detekt-rules-documentation/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/DeprecatedBlockTag.kt
+++ b/detekt-rules-documentation/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/DeprecatedBlockTag.kt
@@ -49,10 +49,6 @@ class DeprecatedBlockTag(config: Config = Config.empty) : Rule(config) {
 
     override fun visitDeclaration(dcl: KtDeclaration) {
         super.visitDeclaration(dcl)
-        verify(dcl)
-    }
-
-    fun verify(dcl: KtDeclaration) {
         dcl.docComment?.getAllSections()?.forEach { section ->
             section.findTagsByName("deprecated").forEach { tag ->
                 report(

--- a/detekt-rules-documentation/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormat.kt
+++ b/detekt-rules-documentation/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormat.kt
@@ -33,11 +33,7 @@ class EndOfSentenceFormat(config: Config = Config.empty) : Rule(config) {
 
     override fun visitDeclaration(dcl: KtDeclaration) {
         super.visitDeclaration(dcl)
-        verify(dcl)
-    }
-
-    fun verify(declaration: KtDeclaration) {
-        declaration.docComment?.let {
+        dcl.docComment?.let {
             val text = it.getDefaultSection().getContent()
             if (text.isEmpty() || text.startsWithHtmlTag()) {
                 return
@@ -46,7 +42,7 @@ class EndOfSentenceFormat(config: Config = Config.empty) : Rule(config) {
                 report(
                     CodeSmell(
                         issue,
-                        Entity.from(declaration),
+                        Entity.from(dcl),
                         "The first sentence of this KDoc does not end with the correct punctuation."
                     )
                 )

--- a/detekt-rules-documentation/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/KDocStyle.kt
+++ b/detekt-rules-documentation/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/KDocStyle.kt
@@ -15,8 +15,7 @@ class KDocStyle(config: Config = Config.empty) : MultiRule() {
     )
 
     override fun visitDeclaration(dcl: KtDeclaration) {
-        super.visitDeclaration(dcl)
-        deprecatedBlockTag.verify(dcl)
-        endOfSentenceFormat.verify(dcl)
+        deprecatedBlockTag.visitDeclaration(dcl)
+        endOfSentenceFormat.visitDeclaration(dcl)
     }
 }

--- a/detekt-rules-documentation/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormatSpec.kt
+++ b/detekt-rules-documentation/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormatSpec.kt
@@ -21,9 +21,9 @@ class EndOfSentenceFormatSpec : Spek({
 
         it("reports invalid KDoc endings on objects") {
             val code = """
-            /** Some doc */
-            object Test {
-            }
+                /** Some doc */
+                fun f(x: Int, y: Int, z: Int) = 
+                    if (x == 0) y + z else x + y
             """
             assertThat(subject.compileAndLint(code)).hasSize(1)
         }

--- a/detekt-rules-documentation/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormatSpec.kt
+++ b/detekt-rules-documentation/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormatSpec.kt
@@ -19,7 +19,7 @@ class EndOfSentenceFormatSpec : Spek({
             assertThat(subject.compileAndLint(code)).hasSize(1)
         }
 
-        it("reports invalid KDoc endings on objects") {
+        it("reports invalid KDoc endings on function with expression body") {
             val code = """
                 /** Some doc */
                 fun f(x: Int, y: Int, z: Int) = 

--- a/detekt-rules-documentation/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/KDocStyleSpec.kt
+++ b/detekt-rules-documentation/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/KDocStyleSpec.kt
@@ -1,0 +1,22 @@
+package io.gitlab.arturbosch.detekt.rules.documentation
+
+import io.gitlab.arturbosch.detekt.test.compileAndLint
+import org.assertj.core.api.Assertions.assertThat
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+class KDocStyleSpec : Spek({
+    val subject by memoized { KDocStyle() }
+
+    describe("check referenced multi rule to only lint errors once per case") {
+
+        it("does only lint once") {
+            val code = """
+            /** Some doc */
+            class Test {
+            }
+            """
+            assertThat(subject.compileAndLint(code)).hasSize(1)
+        }
+    }
+})

